### PR TITLE
Statsd integration

### PR DESCRIPTION
**Pull request by DarcyRaynerDD**
### What does this PR do?

Supports the extension via statsd

### CODE REVIEW

File: datadog_lambda/metric.py [hunk 1]
```diff
@@ -10,20 +10,42 @@
...
 ENHANCED_METRICS_NAMESPACE_PREFIX = "aws.lambda.enhanced"
 
 logger = logging.getLogger(__name__)
 
-lambda_stats = ThreadStats()
-lambda_stats.start()
+
+class StatsDWrapper:
+    """
+    Wraps StatsD calls, to give an identical to ThreadStats
```
Comment (nhinsch) [inline datadog_lambda/metric.py:26]: to give an identical what?

File: tests/test_wrapper.py [hunk 2]
```diff
@@ -38,9 +38,9 @@ def setUp(self):
         self.mock_wrapper_lambda_stats = patcher.start()
         self.addCleanup(patcher.stop)
 
-        patcher = patch("datadog_lambda.metric.lambda_metric")
-        self.mock_lambda_metric = patcher.start()
-        self.addCleanup(patcher.stop)
+        # patcher = patch("datadog_lambda.metric.lambda_metric")
+        # self.mock_lambda_metric = patcher.start()
+        # self.addCleanup(patcher.stop)
```
Comment (nhinsch) [inline tests/test_wrapper.py:43]: What's this?

**Review by Czechh — APPROVED** [review]

File: datadog_lambda/extension.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+def flush_extension():
+    try:
+        requests.post(AGENT_URL + FLUSH_PATH, data={})
+    except Exception as e:
+        logger.debug("Failed to flush extension, returned with error %s", e)
+        return False
+    return True
+
+
+use_extension = is_extension_running()
```
Comment (nhinsch) [inline datadog_lambda/extension.py:29]: To me, `use_extension` sounds like a verb, i.e. if I call it, then the extension will be used. But it seems like it represents whether the extension should be used. Maybe rename this to `is_using_extension` or `should_use_extension` or something along those lines?

**Review by nhinsch — APPROVED** [review]
Lgtm, just a few nits

File: tests/test_wrapper.py (hunk 2 above)
Comment (DarcyRaynerDD) [reply tests/test_wrapper.py:43]: We were patching the datadog_lambda.metric.lambda_metric out for testing, but the new code uses it internally, so I unpatched it. I'll remove this snippet.

**Pull request merged**